server/controllers/auth/signup.js

```diff
@@ -1,3 +1,34 @@
-exports.signup = (req, res) => {
-  res.send(' Hello from signup file');
+const { hash } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { signupSchema } = require('./validation/signup-validation');
+const { users: { getUserByEmailOrUsername, addUser } } = require('../../database/queries');
+
+
+exports.signup = (req, res, next) => {
+  const key = process.env.KEY;
+  const {
+    username, email, password, nativeLangId, learningLangId,
+  } = req.body;
+
+  signupSchema.validate({
+    username, email, password, nativeLangId, learningLangId,
+  })
+    .then(() => getUserByEmailOrUsername(email, username))
+    .then(({ rows }) => {
+      if (rows.length !== 0) throw Error('username or email exists');
+    })
+    .then(() => hash(password, 10))
+    .then((hashed) => addUser({
+      username, email, password: hashed, nativeLangId, learningLangId,
+    }))
+    .then(({ rows }) => jwtSign({ userInfo: { username: rows[0].username, Id: rows[0].id } }, key))
+    .then((sigendPayload) => {
+      res.cookie('token', sigendPayload, { maxAge: 86400000 });
+      res.send('Signup success !!');
```

Comment:
It is not a big deal, but in my opinion, it is better to send an object as a response instead of a text. See this: 
```javascript
res.send({isSuccess: true})
```
It will also be easier for you when you check the response in the client-side since you will be checking a boolean value.

Be consistent as well. Agree on a structure for the response object and follow it in all routes.

Reply:
I second this. If you want to pass some text, put it into the object as a 'msg' field
```
res.send({
    isSuccess: true,
    msg: 'Signup successful',
})
```
But I don't think in this case you need a message, just return the flag and client can decide what to display based on the result..